core/src/main/scala/simulacrum/typeclass.scala

```diff
@@ -249,9 +249,12 @@ class TypeClassMacros(val c: Context) {
     }
 
     def adaptMethods(typeClass: ClassDef, tcInstanceName: TermName, tparamName: Name, proper: Boolean, liftedTypeArgs: List[TypeDef]): List[DefDef] = {
-      val typeClassMethods = typeClass.impl.children.collect {
+      import scala.tools.nsc.ast.Trees
+      def matchMethod: PartialFunction[Tree, DefDef] = {
         case m: DefDef if !m.mods.hasFlag(Flag.PRIVATE) && !m.mods.hasFlag(Flag.PROTECTED) => m
+        case t if t.isInstanceOf[Trees#DocDef] => matchMethod(t.asInstanceOf[Trees#DocDef].definition.asInstanceOf[Tree])
```

Comment:
`case t: Trees#DocDef => t.definition.asInstanceOf[Tree]` won't work?

Reply:
Unfortunately if you try to do this compilation error will occur:
```
Error:(256, 23) pattern type is incompatible with expected type;
 found   : scala.tools.nsc.ast.Trees#DocDef
 required: TypeClassMacros.this.c.universe.TreeApi
        case t: Trees#DocDef => matchMethod(t.definition.asInstanceOf[Tree])
```
Trees from reflect API and NSC aren't compatible. Please check previously linked scala issues.